### PR TITLE
Improve net-metering big-m constraint

### LINKED ISSUE #382: better big-M for net metering
The net metering model is off without indicator constraints - the following JSON results in 30 kW of PV, the net metering limit, when Xpress is used and 62 kW when HiGHS is used due to a large default big-M in the interconnect limit.
[nem_q.json](https://github.com/NREL/REopt.jl/files/15179316/nem_q.json)

### PR DESCRIPTION
**Pull request by zolanaj**
### Fixed 
Improved the practical upper bound on interconnect limit for net metering constraints.  Prevents erroneous results in which net metering limit is exceeded and net metering benefit is still gained.

### Added
New tests for net metering added.

one comment: there's a max benefit of 10 times the annual energy charges which is reached if the wholesale rate provided is larger than the energy price.  One question is: should we throw a warning if the results hit this max benefit?

### CODE REVIEW

**Review by adfarth — COMMENTED** [review]
@zolanaj could you add an update to the Changelog please?

**Comment by zolanaj** [comment]
> @zolanaj could you add an update to the Changelog please?

Done in 5f9af2807637a891d36e5baabeef5544d6a8b029

File: src/constraints/electric_utility_constraints.jl [hunk 1]
```diff
@@ -65,8 +65,20 @@ function add_export_constraints(m, p; _n="")
...
             else
+                #leverage max system sizes for interconnect limit size, alternate is max monthly fully-electrified load in kWh
+                max_interconnection_size = minimum([
+                    p.s.electric_utility.interconnection_limit_kw, 
+                    sum(p.max_sizes[t] for t in NEM_techs),
+                    maximum([sum((
+                        p.s.electric_load.loads_kw[ts] + 
+                        p.s.cooling_load.loads_kw_thermal[ts]/p.cop["ExistingChiller"] + 
+                        (p.s.space_heating_load.loads_kw[ts] + p.s.dhw_load.loads_kw[ts] + p.s.process_heat_load.loads_kw[ts]) # assume electric heater with COP of 1 for conversion to electriity
+                    ) for ts in p.s.electric_tariff.time_steps_monthly[m]) for m in p.months
```
Comment (adfarth) [inline src/constraints/electric_utility_constraints.jl:76]: Why not just `for ts in p.time_steps_with_grid` ?

**Comment by zolanaj** [comment]
post-discussion changes: 
- Added a warning if wholesale benefit is at its max (I'll add an issue to see if we want to go "old school" and make the max benefit much bigger)
- Add multiplier of p.hours_per_time_step to the monthly load big-M candidate

**Comment by zolanaj** [comment]
Mac tests seem to be stalling out but things are working fine in the Windows test suite!

**Review by adfarth — APPROVED** [review]
Thanks so much for implementing this! Strange that the Mac tests aren't passing quickly after your test improvements, but it seems like that issue is not directly related to this PR so I am approving

**Pull request merged**